src/test/java/com/nihongo/EmployeeControllerTest.java

```diff
@@ -0,0 +1,187 @@
+        employee.setDob("20-06-1990");
+        employee.setAddress("Bareilly");
+
+        when(employeeService.updateEmployee(any(), any())).thenReturn(employee);
+
+        MockHttpServletResponse response = mockMvc.perform(put("/employees?id=1")
+                .contentType("application/json")
+                .content(objectMapper.writeValueAsString(employeePutRequest))
+                .accept("application/json")).andReturn().getResponse();
+
+
+        Assertions.assertEquals(200, response.getStatus());
+        Map map = objectMapper.readValue(response.getContentAsString(), Map.class);
+        Assertions.assertEquals(1,map.get("id"));
+    }
+
+    @Test
+    void deleteEmployeeTest() throws Exception {
+        EmployeeResponse employeeResponse = new EmployeeResponse();
+        employeeResponse.setId(Long.parseLong("1"));
+        employeeResponse.setUsername("Sidd");
+        employeeResponse.setMail("mohd.shoaib@gmail.com");
+        employeeResponse.setDob("20-06-1990");
+        employeeResponse.setAddress("Bareilly");
+
+        Employee employee = new Employee();
+        employee.setEmployeeId(Long.parseLong("1"));
+        employee.setUsername("Sidd");
+        employee.setMail("mohd.shoaib@gmail.com");
+        employee.setDob("20-06-1990");
+        employee.setAddress("Bareilly");
+
+        doNothing().when(employeeService).deleteEmployee(any());
+
+        MockHttpServletResponse response = mockMvc.perform(delete("/employees?id=1")
+                .contentType("application/json")
+                .content(objectMapper.writeValueAsString(employeeResponse))
+                .accept("application/json")).andReturn().getResponse();
+
+        Assertions.assertEquals(200, response.getStatus());
+
+        String id = response.getContentAsString();
+        Assertions.assertEquals("1",id);
+        //Map map = objectMapper.readValue(response.getContentAsString(), Map.class);
+    }
+
+    //-===================================================================================================================================
+
+    @Test
+    void getEmployeeTest() throws Exception {
+        EmployeeResponse employeeResponse = new EmployeeResponse();
+        employeeResponse.setId(Long.parseLong("1"));
+        employeeResponse.setUsername("sidd");
+        employeeResponse.setMail("mail");
+        employeeResponse.setDob("0-month");
+        employeeResponse.setAddress("add");
+
+        Employee employee = new Employee();
+        employee.setEmployeeId(Long.parseLong("1"));
+        employee.setUsername("sidd");
+        employee.setMail("mail");
+        employee.setDob("0-month");
+        employee.setAddress("add");
+
+
+        when(employeeService.getEmployee(any())).thenReturn(List.of(employee));
+
+        MockHttpServletResponse response = mockMvc.perform(get("/employees?id=1")
+                .contentType("application/json")
+                .content(objectMapper.writeValueAsString(employeeResponse))
```

Comment:
Do you need to send requestbody (employeeResponse) for get Endpoint?
you can remove it, please delete employe test case we did not send request body there

Reply:
deleted this line.